CI reliability: experimental tests consistent failure

`node-tests-experimental` and `browser-tests-experimental` consistently fail with following messages:

```
Error: backwards-compatability/node10/index.ts(1,28): error TS2307: Cannot find module '@opentelemetry/sdk-node' or its corresponding type declarations.
Error: backwards-compatability/node12/index.ts(1,28): error TS2307: Cannot find module '@opentelemetry/sdk-node' or its corresponding type declarations.
Error: backwards-compatability/node8/index.ts(1,28): error TS2307: Cannot find module '@opentelemetry/sdk-node' or its corresponding type declarations.
```

However, I could not reproduce the problem locally. I believe it is related to https://github.com/open-telemetry/opentelemetry-js/commit/b66c650849ed4da78ab7d9acbbb50df6f8d58e37, /cc @dyladan. Retried CI with the full run and it fails consistently.

Links:
- https://github.com/open-telemetry/opentelemetry-js/pull/2506
  - https://github.com/open-telemetry/opentelemetry-js/pull/2506/checks?check_run_id=3747316161
- https://github.com/open-telemetry/opentelemetry-js/pull/2509
  - https://github.com/open-telemetry/opentelemetry-js/pull/2509/checks?check_run_id=3747384240

--- a/.github/workflows/unit-test.yml
+++ b/.github/workflows/unit-test.yml
@@ -103,7 +103,7 @@ jobs:
         with:
           path: |
             experimental/node_modules
-            experimental/packages/*/node_modules
+            experimental/*/*/node_modules
           key: node-tests-experimental-${{ runner.os }}-${{ matrix.node_version }}-${{ hashFiles('experimental/**/package.json') }}
 
       - name: Bootstrap
@@ -143,7 +143,7 @@ jobs:
         with:
           path: |
             experimental/node_modules
-            experimental/packages/*/node_modules
+            experimental/*/*/node_modules
           key: browser-tests-experimental-${{ runner.os }}-${{ matrix.node_version }}-${{ hashFiles('**/package.json') }}
 
       - name: Bootstrap

--- a/.github/workflows/w3c-integration-test.yml
+++ b/.github/workflows/w3c-integration-test.yml
@@ -22,7 +22,6 @@ jobs:
             package-lock.json
             packages/*/package-lock.json
             benchmark/*/package-lock.json
-            backwards-compatability/*/package-lock.json
             metapackages/*/package-lock.json
             packages/*/package-lock.json
             integration-tests/*/package-lock.json

